pkg/model/core/grpcroute.go

```diff
@@ -0,0 +1,302 @@
+package core
+
+import (
+	"context"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gateway_api_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gateway_api_v1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	GrpcRouteType RouteType = "grpc"
+)
+
+type GRPCRoute struct {
+	r gateway_api_v1alpha2.GRPCRoute
+}
+
+func NewGRPCRoute(route gateway_api_v1alpha2.GRPCRoute) *GRPCRoute {
+	return &GRPCRoute{r: route}
+}
+
+func GetGRPCRoute(client client.Client, ctx context.Context, routeNamespacedName types.NamespacedName) (Route, error) {
+	grpcRoute := &gateway_api_v1alpha2.GRPCRoute{}
+	err := client.Get(ctx, routeNamespacedName, grpcRoute)
+	if err != nil {
+		return nil, err
+	}
+	return NewGRPCRoute(*grpcRoute), nil
+}
+
+func ListGRPCRoutes(client client.Client, context context.Context) []Route {
+	routeList := &gateway_api_v1alpha2.GRPCRouteList{}
+	client.List(context, routeList)
```

Comment:
I believe list can return an error, why drop it here?

Reply:
Fair point, fixed next commit!